### PR TITLE
chore!: merge `SecretStore` into `SecretHandler`

### LINKED ISSUE #245: Clean up secrets handling classes
__What should we do?__

Merge the `SecretStore` into the `SecretHandler` and remove it from API.

__Why should we do it?__

Clean up the classes responsible for secret handling.

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

* Fixes the secret replacement routine in the `SecretHandler` so it can be reused by all Connector context implementations easily. The SecretHandler becomes the recommended class to use for replacing secrets.

BREAKING CHANGE: The class `SecretStore` is removed, use the `SecretHandler`.

## Related issues

closes #245

### CODE REVIEW

File: core/src/main/java/io/camunda/connector/impl/secret/SecretHandler.java [hunk 1]
```diff
@@ -41,10 +44,14 @@ public class SecretHandler implements SecretElementHandler, SecretContainerHandl
   protected static final List<Class<?>> PRIMITIVE_TYPES =
       List.of(String.class, Number.class, Boolean.class);
 
-  private final SecretStore secretStore;
+  private static final Pattern SECRET_PATTERN_FULL = Pattern.compile("^secrets\\.(?<secret>\\S+)$");
+  private static final Pattern SECRET_PATTERN_PLACEHOLDER =
+      Pattern.compile("\\{\\{\\s*secrets\\.(?<secret>\\S+?\\s*)}}");
```
Comment (igpetrov) [inline core/src/main/java/io/camunda/connector/impl/secret/SecretHandler.java:49]: Sorry, not a regex expert but I find weird that first `{` are escaped but the last `}` are not. Is it expected?

**Comment by tmetzke** [comment]
@berndruecker, I had to disable some tests due to them using Spring Zeebe for testing. The latest version however still uses the `SecretStore`, which is why the tests cannot startup. Should be solved with #195.

File: core/src/main/java/io/camunda/connector/impl/secret/SecretHandler.java (hunk 1 above)
Comment (tmetzke) [reply core/src/main/java/io/camunda/connector/impl/secret/SecretHandler.java:49]: Yeah, this is actually just a Java regex issue. In a regex, you actually don't need to escape them (but you can). In Java, if a regex String contains `{`, this is considered as starting a quantification like `{3}`. If you want `{` to be taken literally, you have to escape it in Java regexes. For `}`, the Java regexes are clever enough to see there is no `{` starting a quantification, so you don't have to escape them.

**Review by igpetrov — APPROVED** [review]
LGTM

**Pull request merged**